waspc/src/Wasp/Generator/ServerGenerator.hs

```diff
@@ -84,10 +84,13 @@ genDotEnv :: AppSpec -> Generator [FileDraft]
 genDotEnv spec = return $
   case AS.dotEnvFile spec of
     Just srcFilePath ->
-      [ createCopyFileDraft
-          (C.serverRootDirInProjectRootDir </> dotEnvInServerRootDir)
-          srcFilePath
-      ]
+      if not $ AS.isBuild spec
```

Comment:
I believe you can do following, to avoid a tiny bit of duplication and also simplify code a bit:
```hs
  case AS.dotEnvFile spec of
    Just srcFilePath | not (AS.isBuild spec) ->
      [ createCopyFileDraft
          (C.serverRootDirInProjectRootDir </> dotEnvInServerRootDir)
          srcFilePath
      ]
    Nothing -> []
```